lib/components/particle_component.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:ui';
+
+import 'package:flame/components/particles/composed_particle.dart';
+import 'package:flutter/foundation.dart';
+
+import '../time.dart';
+import 'component.dart';
+
+/// A function which returns [Particle] when called
+typedef ParticleGenerator = Particle Function(int);
+
+/// Base class implementing common behavior for all the particles.
+///
+/// Intention is to follow same "Extreme Composability" style
+/// as across the whole Flutter framework, so each type of particle implements
+/// some particular behavior which then could be nested and combined together
+/// to create specifically required experience.
+abstract class Particle extends Component {
+  /// Generates given amount of particles,
+  /// combining them into one [ComposedParticle]
+  /// Useful for procedural particle generation.
+  static Particle generate({
+    int count = 10,
+    @required ParticleGenerator generator,
+    double lifespan,
+    Duration duration,
+  }) {
+    return ComposedParticle(
+      lifespan: lifespan,
+      duration: duration,
+      children: List<Particle>.generate(count, generator),
+    );
+  }
+
+  Timer _timer;
+  bool _shouldBeDestroyed = false;
+
+  Particle({
+    /// Particle lifespan in [Timer] format,
+    /// double in seconds with microsecond precision
+    double lifespan,
+
+    /// Another way to set lifespan, using Flutter
+    /// [Duration] class
+    Duration duration,
```

Comment:
I feel funky about having two excluding parameters like this... I'd rather have a single non-optional parameter double lifespan (can be optional with default value) and another named constructor .fromDuration that calls the former converting duration to seconds with micro precision

Reply:
Yeah, I see why it feels that way :smile: 
My main intention was to make it possible to utilize an awesome Dart built-in `Duration`. To be honest, I've really struggled to come up with a more elegant approach which would:
- would allow avoiding too much repetitiveness in `Particle` subclasses
- would work regardless of the subclass signature

Having an additional constructor would make it necessary to repeat the main constructor signature completely, and that's for every subclass of `Particle`, which felt a bit too verbose.
I've also tried to experiment with `dynamic`, but it felt even funkier 😅.

Well, now I think that maybe `Duration` support is not  _that_ needed, tbh. Will remove it

Ok, just removed it, took less than to write this comment :smile: 
However, the default value couldn't be utilized properly due to always passing value from subclass constructors. So will use `lifespan ?? .5`.